File: HiringTracker.API/hrt.api/src/routes/health.js

const { Error, HealthResponse } = require('hrt.dto');

const constants = require('../constants');

function routeHealth(router) {
    router.get('/api/v1/health', (req, res) => { getHealth(req, res); })
}

function getHealth(req, res) {
    try {
        let respBody = new HealthResponse();
        respBody.message = "Health is OK";

        res.status(constants.HTTP_OK);
        res.send(respBody);
    }
    catch(error) {
        console.error('Error processing health request', error.message);
        res.status(constants.HTTP_IntServerError);
        let errBody = new Error();
        errBody.message = `Error processing health request: ${error.message}`;
        errBody.code = constants.HTTP_IntServerError;
        res.send(errBody);
    }
}

module.exports = routeHealth;

File: Sources/HiringTracker.API/HRT.DTO/Health.cs
using System;
using System.Collections.Generic;
using System.Text;
using System.Text.Json.Serialization;

namespace HRT.DTO
{
    public class HealthResponse
    {
        [JsonPropertyName("Message")]
        public string Message
        {
            get;
            set;
        }
    }
}
